Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -895,6 +895,7 @@ internal struct BinaryDecoder: Decoder {
 
         if let k = k, let v = v {
             value[k] = v
+            consumed = true
```

Comment:
Right now the code makes both Key and Value optional, so unless we decode them we don't assign into the map.  I'll keep digging for the C++ side.